dbms/src/Interpreters/ExpressionAnalyzer.cpp

```diff
@@ -1040,7 +1040,28 @@ void ExpressionAnalyzer::collectUsedColumns()
 
     /// You need to read at least one column to find the number of rows.
     if (select_query && required.empty())
-        required.insert(ExpressionActions::getSmallestColumn(source_columns));
+    {
+        /// We will find a column with minimum compressed size. Because it is the column that is cheapest to read.
```

Comment:
Could we extract this logic into separate function? A good function name is better then the best of the best comment :)

Reply:
Not sure if worth it for now. I guess we can have a dedicated function next time when more sophisticated rules are added.